paywall/src/unlock.js/startup.ts

```diff
@@ -28,23 +28,35 @@ export default function startup(window: UnlockWindow) {
 
   const origin = '?origin=' + encodeURIComponent(window.origin)
 
+  // Loading the data iframe
   const dataIframe = makeIframe(
     window,
     process.env.PAYWALL_URL + '/static/data-iframe.1.0.html' + origin
   )
+  addIframeToDocument(window, dataIframe)
+
+  // Loading the checkout iframe
   const checkoutIframe = makeIframe(
     window,
     process.env.PAYWALL_URL + '/checkout' + origin
   )
-  // TODO: We should not load the iframe for user account is the configuration does not mention it
-  const userAccountsIframe = makeIframe(
-    window,
-    process.env.USER_IFRAME_URL + origin
-  )
-  addIframeToDocument(window, dataIframe)
-  addIframeToDocument(window, userAccountsIframe)
   addIframeToDocument(window, checkoutIframe)
 
+  // Loading the user account iframe (if applicable)
+  let userAccountsIframe
+  if (normalizedConfig.unlockUserAccounts) {
+    userAccountsIframe = makeIframe(
+      window,
+      process.env.USER_IFRAME_URL + origin
+    )
+  } else {
+    userAccountsIframe = makeIframe(
+      window,
+      '' // We make an "empty" accounts iframe
+    )
+  }
+  addIframeToDocument(window, userAccountsIframe)
```

Comment:
this PR breaks because it changed the order that the iframes are loaded. We should do all of the creation at the top, and call the 3 `addIframeToDocument` where they originally were in the order that is there. Maybe add a comment that the order added is very important?

Also, add a comment that if the order needs to be changed for any reason, then the file `tests/helpers/iframes.js` needs to be changed as well to reflect the new order.

Reply:
I am worried that this code is extremely fragile if the order of iframes actually matters :/

Reply:
It is primarily because it doesn't seem to be possible to figure out which iframe is which in puppeteer (accessing `src` flips out because it's cross-domain), so the integration tests rely on the order being constant. One solution (which just occurred to me) would be to put some kind of data attribute on the iframes to differentiate them. I'll open an issue so I don't forget!